helm-charts/strimzi-kafka-operator/templates/02-role.yaml

```diff
@@ -0,0 +1,178 @@
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: Role
+metadata:
+  name: strimzi-role
```

Comment:
why did you change the `metadata.name` ? I see that you have done something similar for all the others templates, is there any specific reason ?

Reply:
That's a good question.  In the [Review Guidelines](https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md#names-and-labels) they state that charts should have the following metadata for every resource.

> Resources and labels should follow some conventions. The standard resource metadata should be this:
> 
>```
>name: {{ template "myapp.fullname" . }}
>labels:
>  app: {{ template "myapp.name" . }}
>  chart: {{ template "myapp.chart" . }}
>  release: {{ .Release.Name }}
>  heritage: {{ .Release.Service }}
>```

In this case `myapp.fullname` (`strimzi.fullname` in this PR) refers to a templating function in `_helpers.tpl` which concatenates the `Release.name` and `Chart.name`.  I think the reasoning for this is so that you can install multiple instances of the same chart in the same namespace.  I initially applied this convention to all `metadata.name`'s, but then I discovered that the operator depends on certain names to be constant, such as the service account created for the Kafka broker pods.  In the this PR I only use the templated name for the deployment resource. 

I took a look at other charts in kubeapps and noticed a mix of charts enforcing and not enforcing this requirement.  I haven't received an answer yet to my questions about when it's appropriate (i.e. what type of resource in what context) to use the templated name, but I'll follow up by asking on the helm mailing list and get back to you.